src/sentry/tasks/releasemonitor.py

```diff
@@ -182,45 +184,59 @@ def adopt_releases(org_id, totals):
                         / environment_totals["total_sessions"]
                         >= threshold
                     ):
                         try:
                             rpe = ReleaseProjectEnvironment.objects.get(
                                 project_id=project_id,
                                 release_id=Release.objects.get(
-                                    organization=org_id, version=release
+                                    organization=org_id, version=release_version
                                 ).id,
                                 environment__name=environment,
                                 environment__organization_id=org_id,
                             )
                             adopted_ids.append(rpe.id)
                             if rpe.adopted is None:
                                 rpe.update(adopted=timezone.now())
-                        except ReleaseProjectEnvironment.DoesNotExist:
+                        except (Release.DoesNotExist, ReleaseProjectEnvironment.DoesNotExist):
                             metrics.incr("sentry.tasks.process_projects_with_sessions.creating_rpe")
                             try:
-                                env = Environment.objects.get(
+                                env = Environment.objects.get_or_create(
                                     name=environment, organization_id=org_id
-                                )
-                                rel = Release.objects.get(organization=org_id, version=release)
-                                if ReleaseProject.objects.get_or_create(
-                                    project_id=project_id, release=rel
-                                ) and ReleaseEnvironment.objects.get_or_create(
-                                    environment=env, organization_id=org_id, release=rel
-                                ):
-                                    ReleaseProjectEnvironment.objects.create(
-                                        project_id=project_id,
-                                        release_id=rel.id,
-                                        environment=env,
-                                        adopted=timezone.now(),
+                                )[0]
+                                try:
+                                    release = Release.objects.get_or_create(
+                                        organization_id=org_id,
+                                        version=release_version,
+                                        defaults={
+                                            "status": ReleaseStatus.OPEN,
+                                        },
+                                    )[0]
+                                except IntegrityError:
+                                    release = Release.objects.get(
+                                        organization_id=org_id, version=release_version
                                     )
```

Comment:
I just realised this isn't actually needed since we have a lock around release creation: https://github.com/getsentry/sentry/blob/93a8d4e42c3557d6732b1b8c14d1399e8a91e0ca/src/sentry/models/release.py#L521

And actually dug in a bit further, and realised Django does this safely already in `get_or_create` (catches the `IntegrityError` etc), TIL

Reply:
Going to follow up with a cleanup so I don't have to wait for CI again